src/components/featureHighlight/index.js

```diff
@@ -148,11 +148,19 @@ class FeatureHighlight extends BaseComponent {
     this.setTargetPosition();
   }
 
-  UNSAFE_componentWillReceiveProps(nextProps) {
-    this.setTargetPosition(nextProps);
+  static getDerivedStateFromProps(nextProps, prevState) {
+    const target = nextProps?.getTarget?.();
+    const node = FeatureHighlight.findTargetNode(target);
+    if (node && node !== prevState?.node) {
+      return {...prevState, node};
```

Comment:
Why setting all '...prevState' and not only 'node'? What else has changed?

Reply:
`contentPosition`, it works without setting it, but any future state change will be reset in prop change, no?

Reply:
Once a prop changed and `getDerivedStateFromProps` is called you'll get the last state in 'prevState' parameter. Think of this function as `setState()`, you don't need to set the whole state object again, only the part you want to change as a result of a prop change. When you don't return anything from this function, the default is 'null'. Returning null doesn't reset the state, it doesn't touch it...

Reply:
Got it, thanks!